Futures history mobile display

Tburm opened the pull request:
Improve the display of the futures history table on mobile.

## Description
* Implements a specific mobile table for futures history
* Improves types and data handling of futures trade history
* Adds drawer to futures history on mobile
* Improves `PositionType` component to have a mobile version with no background

## Related issue
No issue for the original changes, but follow-up issues:
- #1479 
- #1480

## Screenshots (if appropriate):
<img width="389" alt="image" src="https://user-images.githubusercontent.com/10401554/195420383-2df5f3bc-9882-4539-aba7-4aca2d3b4786.png">
<img width="390" alt="image" src="https://user-images.githubusercontent.com/10401554/195420420-6fe884ee-8da3-47f4-a039-33044ea387f7.png">

avclarke commented on components/Text/PositionType.tsx:
```diff
@@ -34,4 +41,22 @@ const StyledText = styled.p<{ $side: PositionProps['side'] }>`
+	text-transform: uppercase;
+	font-family: ${(props) => props.theme.fonts.bold};
+
+	${(props) =>
```
I think these could be compressed to two lines with:
```
color: ${props => props.side === PositionSide.LONG ? props.theme.colors.common.primaryGreen : props.theme.colors.common. primaryRed}
```

avclarke commented on sections/dashboard/FuturesHistoryTable/FuturesHistoryTable.tsx:
```diff
@@ -22,14 +23,23 @@ import useNetworkSwitcher from 'hooks/useNetworkSwitcher';
 import TimeDisplay from '../../futures/Trades/TimeDisplay';
 
 const FuturesHistoryTable: FC = () => {
+	const [selectedTrade, setSelectedTrade] = React.useState<FuturesTrade>();
```
Be good to stick with avoiding `React.`

avclarke reviewed (commented):
Looks good, just a couple of places for minor clean up.

avclarke commented on components/Text/PositionType.tsx:
```diff
@@ -1,37 +1,57 @@
+	text-transform: uppercase;
+	font-family: ${(props) => props.theme.fonts.bold};
 
 	${(props) =>
```
Also a bit boated here, could reduce to:
```
color: ${(props) => props.side === PositionSide.LONG ? props.theme.colors.common.primaryGreen : props.theme.colors.common.primaryRed}
```

Tburm replied:
fixed this and one other one I hadn't noticed where we repeated the styles for both sides.

avclarke reviewed (dismissed):
Not a blocker but just noticed where the conditional styling is still bloated.

Tburm commented:
@asherism I addressed your comments, but I left the "spot history" table alone right now. I wrote a separate issue to fix that table, since we probably need a specific mobile design. It's doing the same thing running over the side of the container.

asherism commented:
> @asherism I addressed your comments, but I left the "spot history" table alone right now. I wrote a separate issue to fix that table, since we probably need a specific mobile design. It's doing the same thing running over the side of the container.

sounds good. another thing i wanted to mention- we should truncate the 24h vol numbers after 100k or 1m, open interest as well

platschi commented:
> > @asherism I addressed your comments, but I left the "spot history" table alone right now. I wrote a separate issue to fix that table, since we probably need a specific mobile design. It's doing the same thing running over the side of the container.
> 
> sounds good. another thing i wanted to mention- we should truncate the 24h vol numbers after 100k or 1m, open interest as well

Noted, will ticket that out, sounds like a quick devdao ticket. Will test & merge this PR otherwise.
